app/celery/broadcast_message_tasks.py

```diff
@@ -29,7 +32,10 @@ def send_broadcast_event(broadcast_event_id):
 def send_broadcast_provider_message(broadcast_event_id, provider):
     broadcast_event = dao_get_broadcast_event_by_id(broadcast_event_id)
 
-    broadcast_provider_message = create_broadcast_provider_message(broadcast_event, provider)
+    broadcast_provider_message, message_number = create_broadcast_provider_message(broadcast_event, provider)
```

Comment:
i'm not a fan of returning a tuple normally - i think it's often confusing and unexpected, especially when the second value is `None` a lot of the time.

if we add a relationship and an association proxy object linking the ProviderMessage to MessageNumber and vice versa, then we can write `broadcast_provider_message.message_number` to access the value - I've used this pattern recently (look up allowed_broadcast_provider on the service model) and it seems to work well - i'm happy to talk through it with you

Reply:
We did this now ☺️